### PR TITLE
Add Hubert Transfomer to the pipeline

### PR DESCRIPTION
**Pull request by AgaMiko**
- temporary fix of leaf gradient error
- minor fixes
- format code

### CODE REVIEW

File: models/transformers/hubert_transformer.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        self.__output_features = output_features
+
+    def forward(self, input: torch.Tensor, **kwargs):
+        if len(input.shape) > 2:
+            x = torch.reshape(input, (input.shape[0], -1))
+        else:
+            x = input
+        x = self.model(x).last_hidden_state
+        x = torch.reshape(x, (x.shape[0], -1))
+        x = nn.Linear(in_features=x.shape[1], out_features=self.__output_features)(x.cpu())
```
Comment (AgaMiko) [inline models/transformers/hubert_transformer.py:38]: moved x tensor to cpu to avoid cpu/cuda issues when training with GPU.

File: models/transformers/hubert_transformer.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    ):
+        """
+        Args:
+            input_features (int): Argument to keep name convention consistency.
+            output_features (int): Expected number of output features from the model.
+        """
+        super(HubertTransformer, self).__init__()
+        configuration = HubertConfig()
+        self.model = HubertModel(configuration)
+        self.model.feature_extractor._requires_grad = (
```
Comment (AgaMiko) [inline models/transformers/hubert_transformer.py:25]: temporary fix of `HubertModel ` error with gradients

File: models/model_loader.py [hunk 3]
```diff
@@ -13,7 +13,10 @@ class ModelLoader:
         "resnet50_extractor": feature_extractors.Resnet50Extractor,
     }
 
-    transformers_ = {"vanilla_transformer": transformers.VanillaTransformer}
+    transformers_ = {
+        "vanilla_transformer": transformers.VanillaTransformer,
+        "hubert_transformer": transformers.HubertTransformer,
```
Comment (AgaMiko) [inline models/model_loader.py:18]: renamed `PretrainedTransformer` to `HubertTransformer`

File: train.py [hunk 4]
```diff
@@ -129,14 +129,15 @@ def main(args):
...
+        transformer_name="hubert_transformer",
         model_save_dir=args.save,
     )
 
     # TODO - create NeptuneLogger
 
     trainer = pl.Trainer(
         max_epochs=args.epochs,
-        val_check_interval=0.3,
+        val_check_interval=1.0,
```
Comment (AgaMiko) [inline train.py:140]: run `validation_epoch_end` only once every epoch

**Comment by AgaMiko** [comment]
I added more images to assets folder. I resized images to 100x100 to take less space.

File: models/transformers/hubert_transformer.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(
+        self, input_features: int = 2048, output_features: int = 512, *args, **kwargs
+    ):
+        """
+        Args:
+            input_features (int): Argument to keep name convention consistency.
+            output_features (int): Expected number of output features from the model.
+        """
+        super(HubertTransformer, self).__init__()
+        configuration = HubertConfig()
```
Comment (AgaMiko) [inline models/transformers/hubert_transformer.py:23]: added config instead of pretrained for faster run as it was pretrained for audio anyway

**Review by mariaogryczak — APPROVED** [review]

**Review by wiktorfilipiuk — APPROVED** [review]
